testsuite/classlibrary/TestEvent.sc

```diff
@@ -34,6 +34,50 @@ TestEvent : UnitTest {
 
 	}
 
+	test_parent_proto {
```

Comment:
if this test really needs to be this complicated, it would help to have an ASCII diagram or some other sort of description in the code, and not have to read the test asserts to reconstruct it